src/components/Search/Search.js

```diff
@@ -1,29 +1,100 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import SearchIcon from '@material-ui/icons/Search';
 import InputBase from '@material-ui/core/InputBase';
-
 import { FormattedMessage } from 'react-intl';
+import { Link, useHistory } from 'react-router-dom';
 import useStyles from './searchStyles';
+import fuse from './FuseSearch';
 
 function Search() {
   const classes = useStyles();
+  const history = useHistory();
+  const [searchQuery, setSearchQuery] = useState('');
+  const [showResults, setShowResults] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+  const handleChange = event => {
+    setSearchQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    const results = fuse.search(searchQuery).slice(0, 15);
+    setSearchResults(results);
+  }, [searchQuery]);
+
+  const handleSearch = e => {
+    e.preventDefault();
+    const potentialCategory = searchResults.length ? searchResults[0].item.category : '';
+    setShowResults(false);
+    history.push(`/search?q=${searchQuery}&c=${potentialCategory}&pr=0-0`);
+  };
+
+  const handleRoute = to => {
+    history.push(to.pathname);
+  };
+
   return (
-    <div className={classes.search}>
+    <div
+      onFocus={() => setShowResults(true)}
+      onBlur={() => setShowResults(false)}
+      className={classes.search}
+    >
       <div className={classes.searchIcon}>
         <SearchIcon />
       </div>
-      <FormattedMessage id="search">
-        {placeHolderText => (
-          <InputBase
-            placeholder={`${placeHolderText}`}
-            classes={{
-              root: classes.inputRoot,
-              input: classes.inputInput,
-            }}
-            inputProps={{ 'aria-label': 'search' }}
-          />
-        )}
-      </FormattedMessage>
+      <form className={classes.root} noValidate autoComplete="off" onSubmit={handleSearch}>
+        <FormattedMessage id="search">
+          {placeHolderText => (
+            <InputBase
+              value={searchQuery}
+              onChange={handleChange}
+              placeholder={`${placeHolderText}`}
+              classes={{
+                root: classes.inputRoot,
+                input: classes.inputInput,
+              }}
+              className="searchField"
+              inputProps={{ 'aria-label': 'search' }}
+            />
+          )}
+        </FormattedMessage>
+      </form>
+      <div
+        className={`position-absolute searchResultContainer ${
```

Comment:
Keep things consistent

```suggestion
        className={`position-absolute search-result-container ${
```